sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala

```diff
@@ -285,334 +290,86 @@ class TableIdentifierParserSuite extends SparkFunSuite with SQLHelper {
+        "apache", "spark", "sql", "catalyst", "parser", "SqlBase.g4").toFile
+    }
+    fileToString(sqlBasePath).split("\n")
+  }
 
-  val reservedKeywordsInAnsiMode = Set(
-    "all",
-    "and",
-    "anti",
-    "any",
-    "as",
-    "authorization",
-    "both",
-    "case",
-    "cast",
-    "check",
-    "collate",
-    "column",
-    "constraint",
-    "create",
-    "cross",
-    "current_date",
-    "current_time",
-    "current_timestamp",
-    "current_user",
-    "day",
-    "distinct",
-    "else",
-    "end",
-    "escape",
-    "except",
-    "false",
-    "fetch",
-    "for",
-    "foreign",
-    "from",
-    "full",
-    "grant",
-    "group",
-    "having",
-    "hour",
-    "in",
-    "inner",
-    "intersect",
-    "into",
-    "join",
-    "is",
-    "leading",
-    "left",
-    "minute",
-    "month",
-    "natural",
-    "not",
-    "null",
-    "on",
-    "only",
-    "or",
-    "order",
-    "outer",
-    "overlaps",
-    "primary",
-    "references",
-    "right",
-    "select",
-    "semi",
-    "session_user",
-    "minus",
-    "second",
-    "some",
-    "table",
-    "then",
-    "to",
-    "trailing",
-    "union",
-    "unique",
-    "unknown",
-    "user",
-    "using",
-    "when",
-    "where",
-    "with",
-    "year")
+  private def parseAntlrGrammars(startTag: String, endTag: String)
+      (f: PartialFunction[String, Option[String]]): Set[String] = {
+    // We need to map a symbol string to actual literal strings
+    // in case that they have different strings.
+    val symbolsToNeedRemap = Map(
+      "DATABASES" -> Seq("DATABASES", "SCHEMAS"),
+      "RLIKE" -> Seq("RLIKE", "REGEXP"),
+      "SETMINUS" -> Seq("MINUS"),
+      "TEMPORARY" -> Seq("TEMPORARY", "TEMP")
+    )
+    val keywords = new mutable.ArrayBuffer[String]
+    val default = (_: String) => None
+    var startTagFound = false
+    var parseFinished = false
+    val lineIter = sqlSyntaxDefs.toIterator
+    while (!parseFinished && lineIter.hasNext) {
+      val line = lineIter.next()
+      if (line.trim.startsWith(startTag)) {
+        startTagFound = true
+      } else if (line.trim.startsWith(endTag)) {
+        parseFinished = true
+      } else if (startTagFound) {
+        f.applyOrElse(line, default).foreach { symbol =>
+          if (symbolsToNeedRemap.contains(symbol)) {
+            keywords ++= symbolsToNeedRemap(symbol)
+          } else {
+            keywords += symbol
+          }
+        }
+      }
+    }
+    assert(keywords.nonEmpty && startTagFound && parseFinished, "cannot extract keywords from " +
+      s"the `SqlBase.g4` file, so please check if the start/end tags (`$startTag` and `$endTag`) " +
+      "are placed correctly in the file.")
+    keywords.map(_.trim.toLowerCase(Locale.ROOT)).toSet
+  }
 
-  val nonReservedKeywordsInAnsiMode = allCandidateKeywords -- reservedKeywordsInAnsiMode
+  private def parseSyntax(startTag: String, endTag: String): Set[String] = {
```

Comment:
nit: it's only used once, can we inline it?

Reply:
Ah, right. okay.